vmm/src/device_manager.rs

```diff
@@ -3405,6 +3412,27 @@ impl DeviceManager {
 
         counters
     }
+
+    pub fn resize_balloon(&mut self, size: u64) -> DeviceManagerResult<()> {
```

Comment:
This part is not right.
```
    pub fn balloon_resize(&mut self, expected_ram: u64) -> Result<u64, Error> {
        let mut balloon_size = 0;
        if let Some(balloon) = &self.balloon {
            if expected_ram < self.current_ram {
                balloon_size = self.current_ram - expected_ram;
            }
            balloon
                .lock()
                .unwrap()
                .resize(balloon_size)
                .map_err(Error::VirtioBalloonResizeFail)?;
        }

        Ok(balloon_size)
    }
```

Reply:
I know and I changed the behavior on purpose. I think this is more straightforward to set the balloon size. The management layer on top of CH can do the math as needed.

Reply:
Current control interface of balloon on qemu is set the ram size with balloon.
I'm worried that people won't understand how balloon size of clh is set up.

Reply:
Yes but this is confusing, so I really prefer having the user setting the balloon size.

Reply:
If so, the people that use the balloon must understand the core design of balloon.

I am OK with the control interface set the balloon size if the help introduce it clear.